internal/pkg/agent/cmd/enroll_match_fileowner_unix.go

```diff
@@ -0,0 +1,63 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+//go:build !windows
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"syscall"
+)
+
+func getFileOwner(filePath string) (string, error) {
+	fileInfo, err := os.Stat(filePath)
+	if err != nil {
+		return "", fmt.Errorf("failed to get file info: %w", err)
+	}
+
+	stat, ok := fileInfo.Sys().(*syscall.Stat_t)
+	if !ok {
+		return "", fmt.Errorf("failed to get system specific file info: %w", err)
+	}
+	return strconv.FormatInt(int64(stat.Uid), 10), nil
+}
+
+func getCurrentUser() (string, error) {
+	return strconv.Itoa(os.Geteuid()), nil
+}
+
+func isFileOwner(curUser string, fileOwner string) (bool, error) {
+	return curUser == fileOwner, nil
+}
+
+func isEnroll() (bool, error) {
```

Comment:
i don't think `isEnroll` is a proper name for this. not sure it captures properly what this func is doing

Reply:
I can also split this function into its constituent functions. It would be more readable. All the steps would be clearly outlined. I couldn't come up with a reasonably short function name that describes exactly what's happening.

If I split it, these functions will be called in order in enroll.go
- getFileOwner
- getCurrentUser
- isFileOwner

Reply:
I agree with @michalpristas here, this is not the proper name for this function. It is just verifying that the executing users is the same as the owner of the executable.

To me the better name would be `isOwnerExec`?